snapcraft/internal/lxd/_containerbuild.py

```diff
@@ -76,6 +79,93 @@ def __init__(self, *, output, source, project_options,
             os.path.join('~', 'snap', 'lxd', 'common'))
         os.makedirs(self._lxd_common_dir, exist_ok=True)
 
+    def _get_multipass_remote(self):
+        try:
+            containers = subprocess.check_output([
+                'multipass', 'list']).decode()
+            # Trailing space to disambiguate names starting with snapcraft
+            if 'snapcraft ' not in containers:
+                subprocess.check_call([
+                    'multipass', 'launch', '--name=snapcraft', '--disk', '10G',
+                    ])
+            subprocess.check_call([
+                'multipass', 'start', 'snapcraft'])
+            self._setup_lxd()
+
+            return 'multipass'
+        except FileNotFoundError:
+            raise ContainerError('Multipass is not installed')
+        except subprocess.CalledProcessError as e:
+            raise ContainerError(
+                'Failed to setup multipass remote: {!r}'.format(
+                    ' '.join(e.cmd)))
+
+    def _setup_lxd(self):
+        if 'lxd' not in subprocess.check_output([
+                'multipass', 'exec', 'snapcraft', '--',
+                'snap', 'list']).decode():
+            logger.debug('Installing LXD')
+            subprocess.check_call([
+                'multipass', 'exec', 'snapcraft', '--',
+                'sudo', 'snap', 'install', 'lxd'])
+        subprocess.check_call([
+            'multipass', 'exec', 'snapcraft', '--',
+            'sudo', '/snap/bin/lxd', 'waitready'])
+        if 'core.trust_password: true' not in subprocess.check_output([
+                'multipass', 'exec', 'snapcraft', '--',
+                '/snap/bin/lxc', 'config', 'show']).decode():
+            logger.debug('Configuring LXD')
+            # Expose LXD over the network
+            subprocess.check_call([
+                'multipass', 'exec', 'snapcraft', '--',
+                'sudo', '/snap/bin/lxd', 'init', '--auto',
+                '--network-address', '0.0.0.0',
+                '--network-port', '8443',
+                '--trust-password', 'snapcraft'])
+        if 'lxdbr0' not in subprocess.check_output([
+                'multipass', 'exec', 'snapcraft', '--',
+                '/snap/bin/lxc', 'network', 'list']).decode():
+            subprocess.check_call([
+                'multipass', 'exec', 'snapcraft', '--',
+                'sudo', '/snap/bin/lxc', 'network',
+                'create', 'lxdbr0'])
+            subprocess.check_call([
+                'multipass', 'exec', 'snapcraft', '--',
+                'sudo', '/snap/bin/lxc', 'network',
+                'attach-profile', 'lxdbr0', 'default', 'eth0'])
+
+        remotes = subprocess.check_output([
+            'lxc', 'remote', 'list']).decode()
```

Comment:
What? So multipass is actually a lxd remote? crazy, disregard my previous comment.

Reply:
lxd is installed in the multipasss environment and setup to work as a remote.